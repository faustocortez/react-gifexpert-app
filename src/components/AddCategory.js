import React, { useState } from 'react'
import PropTypes  from 'prop-types';

export const AddCategory = ({ setCategories }) => {
    
    const [inputValue, setInputValue] = useState(''); // initialze value to aviod undefinied
    
    const handleInputChange = (e) => {
        const text = e.target.value;
        setInputValue(text);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        

        if (inputValue.trim().length > 2) {
            // We can use the previos data in categories only passing the setCategory function
            setCategories(categories => [inputValue, ...categories]);
            setInputValue('');
        }
    }

    return (
        <form onSubmit={ handleSubmit }>
            <p> { inputValue } </p>
            <input
                type="text"
                onChange={ handleInputChange }
                value={inputValue}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
}
