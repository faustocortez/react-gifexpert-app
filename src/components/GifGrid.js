import React, { useState, useEffect } from 'react'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([]);

    // component did mount in react classes
    // the second parameter "[]" is a list of dependencies
    // if [] equal to one execution
    useEffect(() => {
        getGifs();
    }, []);

    const getGifs = async() => {
        const url = 'https://api.giphy.com/v1/gifs/search?api_key=LorHcp1hTRLEIFFnzCsCcrXK0nOAS0Ux&q=Rick+and+Morty&limit=10';
        const response = await fetch( url );
        const { data } = await response.json();

        const gifs = data.map(img => {
            const {
                id,
                title,
                images: { downsized_medium: { url} },
            } = img;
            return  {
                id,
                title,
                url
            }
        })

        console.log(gifs);
        setImages(gifs);
    }

    return (
        <>
            <h3>{ category }</h3>
            <div className="card-grid">
                { 
                    images.map( img => (    
                        <GifGridItem
                            key={ img.id }
                            { ...img }
                        />
                    ))
                }
            </div>
        </>
    )
}
