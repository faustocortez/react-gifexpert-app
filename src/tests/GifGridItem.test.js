import react from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../components/GifGridItem';


describe('Pruebas del componente GifGridItem', () => {
    // Para actualizar un snapshot presionamos "u" en consola.
    // De esta manera podemos podemos validar que nuestras condiciones
    // funcionen como esperamos evaluando un antes y después del snapshot.

    const title = 'Un título';
    const url = 'https://localhost';
    const wrapper = shallow(<GifGridItem title={ title } url={ url } />);

    test('* debe mostrar el component GifGridItem correctamente', () => {

        expect(wrapper).toMatchSnapshot();
    });

    // Evaluando el <p>párrafo</p> del componente
    test('* debe de tener un párrafo con el title', () => {
        // el wrapper funciona como el query selector del DOM en JS
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });
    
    test('* debe de tener la imágen al url y al alt de los props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);

        // para ver el html
        // console.log(img.html());

        // para ver las props del elemento
        // console.log(img.props());
    });

    test('* debe de tener la clase animate__fadeIn', () => {
        const div = wrapper.find('div');
        const animateClass = 'animate__fadeIn';

        expect(div.hasClass(animateClass)).toEqual(true);
    })
    
    
    
});