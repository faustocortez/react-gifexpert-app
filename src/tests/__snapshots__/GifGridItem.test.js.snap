// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente GifGridItem * debe mostrar el component GifGridItem correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título"
    src="https://localhost"
  />
  <p>
    Un título
  </p>
</div>
`;
